src/common-ssh/ssh.c

```diff
@@ -518,6 +520,71 @@ guac_common_ssh_session* guac_common_ssh_create_session(guac_client* client,
         return NULL;
     }
 
+    /* SSH known host key checking. */
+    LIBSSH2_KNOWNHOSTS *ssh_known_hosts = libssh2_knownhost_init(session);
+    int num_known_hosts = 0;
+
+    /* Add host key provided from settings */
+    if (host_key && strcmp(host_key, "") != 0) {
+
+        int kh_add = libssh2_knownhost_readline(ssh_known_hosts, host_key, strlen(host_key),
+                LIBSSH2_KNOWNHOST_FILE_OPENSSH);
+        num_known_hosts++;
+
+        if (kh_add)
+            guac_client_log(client, GUAC_LOG_WARNING, "Failed to add provided host key"
+                    " to known hosts store for %s.  Error was %d", hostname, kh_add);
```

Comment:
Is there any way to get a human-readable error? The integer value of `kh_add` isn't likely to have any meaning to the user reading the logs, let alone to us when reported as a bug.